src/storage-client/src/sink.rs

```diff
@@ -390,21 +390,18 @@ pub async fn determine_latest_progress_record(
     where
         C: ConsumerContext,
     {
-        if let Some(result) = consumer.poll(timeout) {
-            match result {
-                Ok(message) => match message.payload() {
-                    Some(p) => Ok(Some((
-                        message.key().unwrap_or(&[]).to_vec(),
-                        p.to_vec(),
-                        message.offset(),
-                    ))),
-                    None => bail!("unexpected null payload"),
-                },
-                Err(KafkaError::PartitionEOF(_)) => Ok(None),
-                Err(err) => bail!("Failed to process message {}", err),
-            }
-        } else {
-            Ok(None)
+        match consumer.poll(timeout) {
+            Some(Ok(message)) => match message.payload() {
+                Some(p) => Ok(Some((
+                    message.key().unwrap_or(&[]).to_vec(),
+                    p.to_vec(),
+                    message.offset(),
+                ))),
+                None => bail!("unexpected null payload"),
+            },
+            Some(Err(KafkaError::PartitionEOF(_))) => Ok(None),
```

Comment:
FWIW, I'm not sure how much to trust this mechanism... it's not really supported at the protocol level.

I don't think false positives here have real bad outcomes, since we check that we see the highwatermark below. But in that case I'm not sure why we don't just treat this like a `None` result.